source/common/config/grpc_mux_subscription_impl.cc

```diff
@@ -22,8 +22,12 @@ GrpcMuxSubscriptionImpl::GrpcMuxSubscriptionImpl(GrpcMux& grpc_mux,
 void GrpcMuxSubscriptionImpl::start(const std::set<std::string>& resources) {
   if (init_fetch_timeout_.count() > 0) {
     init_fetch_timeout_timer_ = dispatcher_.createTimer([this]() -> void {
-      ENVOY_LOG(warn, "gRPC config: initial fetch timed out for {}", type_url_);
-      callbacks_.onConfigUpdateFailed(nullptr);
+      try {
+        throw EnvoyException("initial fetch timed out");
```

Comment:
You should be able to allocate an EnvoyException on the stack and pass it into the update failed function.